### PR TITLE
Delegation transactions outcome parser

### PR DESCRIPTION
**Pull request by popenta**
Implemented the delegation transactions parser according to the [specs](https://github.com/multiversx/mx-sdk-specs/blob/main/core/transactions-outcome-parsers/delegation_transactions_outcome_parser.md).

### CODE REVIEW

File: multiversx_sdk/__init__.py [hunk 1]
```diff
@@ -65,5 +68,6 @@
     "GenericError", "GenericResponse", "ApiNetworkProvider", "ProxyNetworkProvider",
     "UserSigner", "Mnemonic", "UserSecretKey", "UserPublicKey", "ValidatorSecretKey",
     "ValidatorPublicKey", "UserVerifier", "ValidatorSigner", "ValidatorVerifier", "ValidatorPEM",
-    "UserWallet", "UserPEM", "QueryRunnerAdapter", "TransactionsConverter"
+    "UserWallet", "UserPEM", "QueryRunnerAdapter", "TransactionsConverter", "DelegationTransactionsOutcomeParser",
+    "find_events_by_identifier"
```
Comment (andreibancioiu) [inline multiversx_sdk/__init__.py:72]: They seem to be only partially sorted.

**Review by bogdan-rosianu — APPROVED** [review]

File: multiversx_sdk/core/transactions_outcome_parsers/delegation_transactions_outcome_parser.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def parse_create_new_delegation_contract(
+            self,
+            transaction_outcome: TransactionOutcome
+    ) -> List[CreateNewDelegationContractOutcome]:
+        self._ensure_no_error(transaction_outcome.logs.events)
+
+        events = find_events_by_identifier(transaction_outcome, "SCDeploy")
+        return [CreateNewDelegationContractOutcome(self._extract_contract_address(event)) for event in events]
+
+    def _ensure_no_error(self, transaction_events: List[TransactionEvent]) -> None:
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transactions_outcome_parsers/delegation_transactions_outcome_parser.py:25]: Is duplicated now, we should think of a refactoring (for a future PR).

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk/__init__.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk/__init__.py:72]: The `__all__` list is not sorted at all.

File: multiversx_sdk/core/transactions_outcome_parsers/delegation_transactions_outcome_parser.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk/core/transactions_outcome_parsers/delegation_transactions_outcome_parser.py:25]: I was thinking the same thing. Will do in the next PR.

**Pull request merged**